### PR TITLE
Feature/173964655 add proteomics differential experiment type

### PR DESCRIPTION
**Pull request by lingyun1010**
This PR comes with the bulk branch https://github.com/ebi-gene-expression-group/atlas-web-bulk/tree/feature%2F173964655-add-proteomics-differential

### CODE REVIEW

**Comment by upendrakumbham** [comment]
I'm getting the above issue when trying to see the experiments page. But when I hit `http://localhost:8080/gxa/json/experiments/E-PROT-39` endpoint, I could see the JSON payload for the experiment: E-PROT-39.

File: src/main/java/uk/ac/ebi/atlas/experimentpage/ExperimentDownloadSupplier.java [hunk 1]
```diff
@@ -298,8 +299,14 @@ protected void write(Writer responseWriter,
         @Override
         public Collection<ExternallyAvailableContent> get(DifferentialExperiment experiment) {
             DifferentialRequestPreferences preferences = new DifferentialRequestPreferences();
-            preferences.setFoldChangeCutoff(0.0);
-            preferences.setCutoff(1.0);
+            if (experiment.getType().isProteomicsDifferential()) {
+                preferences.setFoldChangeCutoff(0.0);
+                preferences.setCutoff(0.5);
```
Comment (alfonsomunozpomer) [inline src/main/java/uk/ac/ebi/atlas/experimentpage/ExperimentDownloadSupplier.java:304]: Have you verified with curators that this is the default values we want?

File: src/main/java/uk/ac/ebi/atlas/profiles/stream/RnaSeqAndProteomicsProfileStreamFactory.java [hunk 2]
```diff
@@ -18,7 +18,7 @@
 import java.util.function.Predicate;
 
 @Named
-public class RnaSeqProfileStreamFactory
+public class RnaSeqAndProteomicsProfileStreamFactory
         extends ProfileStreamFactory<Contrast,
```
Comment (alfonsomunozpomer) [inline src/main/java/uk/ac/ebi/atlas/profiles/stream/RnaSeqAndProteomicsProfileStreamFactory.java:22]: Notice that `RnaSeqAndProteomicsProfileStreamFactory` is extending a `ProfileStreamFactory` of `RnaSeqRequestContext` and `RnaSeqProfile`, which kind of breaks the semantics of the class hierarchy.

File: src/main/java/uk/ac/ebi/atlas/profiles/writer/RnaSeqAndProteomicsDifferentialProfilesWriterFactory.java [hunk 3]
```diff
@@ -7,6 +7,6 @@
 import javax.inject.Named;
 
 @Named
-public class RnaSeqDifferentialProfilesWriterFactory extends
+public class RnaSeqAndProteomicsDifferentialProfilesWriterFactory extends
```
Comment (alfonsomunozpomer) [inline src/main/java/uk/ac/ebi/atlas/profiles/writer/RnaSeqAndProteomicsDifferentialProfilesWriterFactory.java:10]: Similar as above, `RnaSeqAndProteomicsDifferentialProfilesWriterFactory` extends a writer factory for `RnaSeqProfile` and `RnaSeqRequestContext`.

**Review by alfonsomunozpomer — CHANGES_REQUESTED** [review]
As mentined in my previous review, generalising the stream and writer factory requires also thought to rename their generic data types.

Also, going back to basic fundamentals of software engineering (and system design), it’s a bad habit to generalise a class or component by enumerating its implemented types.

For example, let’s say I have a `Koala` class:
```java
public class Koala {
  public void breathes(){};
  public void eatsEukalyptus(){};
}
``` 

And then I add a new class, `Sloth`:
```java
public class Sloth {
  public void breathes(){};
  public void eatsTenderShoots(){};
}
```

The right way to generalise these two types is:
```java
public class ArborealAnimal {
  public void breathes(){};
}
```

```java
public class Koala extends Animal {
  public void eatsEukalyptus(){};
}
```

```java
public class Sloth extends Animal {} {
  public void eatsTenderShoots(){};
}
```

If you apply the logic applied in this PR we would have something like:
```java
public class KoalaAndSloth {
  public void breathes(){};

  public void eats(){
    // Eukalyptus if it’s a koala
    // Tender shoots if it’s a sloth
  };
}
```

So, if you generalise a class the name should also be general, rather than enumerating the real world objects it maps to. You can see how the second approach becomes less manageable as more animals are added to the system.

File: src/main/java/uk/ac/ebi/atlas/experimentpage/ExperimentDownloadSupplier.java [hunk 4]
```diff
@@ -18,14 +18,10 @@
 import uk.ac.ebi.atlas.model.experiment.differential.microarray.MicroarrayExperiment;
 import uk.ac.ebi.atlas.profiles.ProfileStreamFilter;
 import uk.ac.ebi.atlas.profiles.baseline.BaselineProfileStreamOptions;
-import uk.ac.ebi.atlas.profiles.stream.MicroarrayProfileStreamFactory;
-import uk.ac.ebi.atlas.profiles.stream.ProfileStreamFactory;
-import uk.ac.ebi.atlas.profiles.stream.ProteomicsBaselineProfileStreamFactory;
-import uk.ac.ebi.atlas.profiles.stream.RnaSeqBaselineProfileStreamFactory;
-import uk.ac.ebi.atlas.profiles.stream.RnaSeqAndProteomicsProfileStreamFactory;
+import uk.ac.ebi.atlas.profiles.stream.*;
```
Comment (alfonsomunozpomer) [inline src/main/java/uk/ac/ebi/atlas/experimentpage/ExperimentDownloadSupplier.java:21]: Expand wildcard imports, please.

**Review by alfonsomunozpomer — APPROVED** [review]
So much better! Actually I think `BulkDifferential...` is a much better name than `RnaSeq...` because it better represents more accurately what these classes are about. Good choice.

The only thing remaining is to rename the test class `RnaSeqProfileStreamFactoryPickUpExpressionsByIndexTest` to `BulkDifferentialProfileStreamFactoryPickUpExpressionsByIndexTest` and all good.

**Pull request merged**